Reference i?amin produce unexpected results with NaN input

# Summary

When compiled with recent dpcpp compilers, the `cblas_i?amin` routines in `src/blas/backends/netlib/netlib_level1.cpp` and the `iamin` routines in `tests/unit_tests/blas/include/reference_blas_templates.hpp` produce incorrect results when there are `NaN`s in the input, because `std::isnan` always returns true for dpcpp by default unless you use `-fp-model=precise`.

# Version

Current `develop` branch.

# Steps to reproduce

Using the `cblas_isamin` function from `netlib_level1.cpp`, compile and run the following test:

```c++
int main(int argc, char * argv[]) {
    float * x = (float*) malloc(3 * sizeof(float));
    int idx = 0;

    auto check_three = [&](float a, float b, float c, int expected) -> int {
        x[0] = a; x[1] = b; x[2] = c;
        idx = cblas_isamin(3, x, 1);
        std::cout << "    idx: " << idx << ", expected: " << expected << std::endl;
        if (idx != expected) return 1;
        return 0;
    };

    int result = 0;
    result += check_three(fnan, -0.5, 1.0, 0);
    result += check_three(-0.5, fnan, 1.0, 1);
    result += check_three(-0.5, 1.0, fnan, 2);
    result += check_three(0.0, fnan, 1.0, 1);
    result += check_three(1.0, fnan, 0.0, 1);
    result += check_three(-0.3, 2.1, fnan, 2);
    result += check_three(2.1, -0.3, fnan, 2);
    result += check_three(fnan, -0.3, 2.1, 0);
    result += check_three(fnan, fnan, 1.0, 0);
    result += check_three(fnan, 1.0, fnan, 0);
    result += check_three(1.0, fnan, fnan, 1);
    if (result) std::cout << "  FAILED!\n"; 

    free(x);
    return (result != 0);
}
```

# Observed/expected behavior

Output of build and run:
```
issue.cpp:17:29: warning: comparison with NaN always evaluates to false in fast floating point modes [-Wtautological-constant-compare]
        bool is_first_nan = std::isnan(curr_val) && !std::isnan(min_val);
                            ^~~~~~~~~~~~~~~~~~~~
issue.cpp:17:54: warning: comparison with NaN always evaluates to false in fast floating point modes [-Wtautological-constant-compare]
        bool is_first_nan = std::isnan(curr_val) && !std::isnan(min_val);
                                                     ^~~~~~~~~~~~~~~~~~~
2 warnings generated.
    idx: 0, expected: 0
    idx: 0, expected: 1
    idx: 0, expected: 2
    idx: 0, expected: 1
    idx: 2, expected: 1
    idx: 0, expected: 2
    idx: 1, expected: 2
    idx: 0, expected: 0
    idx: 0, expected: 0
    idx: 0, expected: 0
    idx: 0, expected: 1
  FAILED!
```

--- a/src/blas/backends/netlib/netlib_level1.cpp
+++ b/src/blas/backends/netlib/netlib_level1.cpp
@@ -45,12 +45,14 @@ int cblas_isamin(int n, const float *x, int incx) {
     }
     int min_idx = 0;
     auto min_val = abs_val(x[0]);
-    if (sycl::isnan(min_val)) return 0;
+    if (sycl::isnan(min_val))
+        return 0;
 
     for (int logical_i = 1; logical_i < n; ++logical_i) {
         int i = logical_i * std::abs(incx);
         auto curr_val = abs_val(x[i]);
-        if (sycl::isnan(curr_val)) return logical_i;
+        if (sycl::isnan(curr_val))
+            return logical_i;
         if (curr_val < min_val) {
             min_idx = logical_i;
             min_val = curr_val;
@@ -65,12 +67,14 @@ int cblas_idamin(int n, const double *x, int incx) {
     }
     int min_idx = 0;
     auto min_val = abs_val(x[0]);
-    if (sycl::isnan(min_val)) return 0;
+    if (sycl::isnan(min_val))
+        return 0;
 
     for (int logical_i = 1; logical_i < n; ++logical_i) {
         int i = logical_i * std::abs(incx);
         auto curr_val = abs_val(x[i]);
-        if (sycl::isnan(curr_val)) return logical_i;
+        if (sycl::isnan(curr_val))
+            return logical_i;
         if (curr_val < min_val) {
             min_idx = logical_i;
             min_val = curr_val;
@@ -85,12 +89,14 @@ int cblas_icamin(int n, const std::complex<float> *x, int incx) {
     }
     int min_idx = 0;
     auto min_val = abs_val(x[0]);
-    if (sycl::isnan(min_val)) return 0;
+    if (sycl::isnan(min_val))
+        return 0;
 
     for (int logical_i = 1; logical_i < n; ++logical_i) {
         int i = logical_i * std::abs(incx);
         auto curr_val = abs_val(x[i]);
-        if (sycl::isnan(curr_val)) return logical_i;
+        if (sycl::isnan(curr_val))
+            return logical_i;
         if (curr_val < min_val) {
             min_idx = logical_i;
             min_val = curr_val;
@@ -105,12 +111,14 @@ int cblas_izamin(int n, const std::complex<double> *x, int incx) {
     }
     int min_idx = 0;
     auto min_val = abs_val(x[0]);
-    if (sycl::isnan(min_val)) return 0;
+    if (sycl::isnan(min_val))
+        return 0;
 
     for (int logical_i = 1; logical_i < n; ++logical_i) {
         int i = logical_i * std::abs(incx);
         auto curr_val = abs_val(x[i]);
-        if (sycl::isnan(curr_val)) return logical_i;
+        if (sycl::isnan(curr_val))
+            return logical_i;
         if (curr_val < min_val) {
             min_idx = logical_i;
             min_val = curr_val;

--- a/src/blas/backends/netlib/netlib_level1.cpp
+++ b/src/blas/backends/netlib/netlib_level1.cpp
@@ -45,12 +45,13 @@ int cblas_isamin(int n, const float *x, int incx) {
     }
     int min_idx = 0;
     auto min_val = abs_val(x[0]);
+    if (sycl::isnan(min_val)) return 0;
 
-    for (int logical_i = 0; logical_i < n; ++logical_i) {
+    for (int logical_i = 1; logical_i < n; ++logical_i) {
         int i = logical_i * std::abs(incx);
         auto curr_val = abs_val(x[i]);
-        bool is_first_nan = std::isnan(curr_val) && !std::isnan(min_val);
-        if (is_first_nan || curr_val < min_val) {
+        if (sycl::isnan(curr_val)) return logical_i;
+        if (curr_val < min_val) {
             min_idx = logical_i;
             min_val = curr_val;
         }
@@ -64,12 +65,13 @@ int cblas_idamin(int n, const double *x, int incx) {
     }
     int min_idx = 0;
     auto min_val = abs_val(x[0]);
+    if (sycl::isnan(min_val)) return 0;
 
-    for (int logical_i = 0; logical_i < n; ++logical_i) {
+    for (int logical_i = 1; logical_i < n; ++logical_i) {
         int i = logical_i * std::abs(incx);
         auto curr_val = abs_val(x[i]);
-        bool is_first_nan = std::isnan(curr_val) && !std::isnan(min_val);
-        if (is_first_nan || curr_val < min_val) {
+        if (sycl::isnan(curr_val)) return logical_i;
+        if (curr_val < min_val) {
             min_idx = logical_i;
             min_val = curr_val;
         }
@@ -83,12 +85,13 @@ int cblas_icamin(int n, const std::complex<float> *x, int incx) {
     }
     int min_idx = 0;
     auto min_val = abs_val(x[0]);
+    if (sycl::isnan(min_val)) return 0;
 
-    for (int logical_i = 0; logical_i < n; ++logical_i) {
+    for (int logical_i = 1; logical_i < n; ++logical_i) {
         int i = logical_i * std::abs(incx);
         auto curr_val = abs_val(x[i]);
-        bool is_first_nan = std::isnan(curr_val) && !std::isnan(min_val);
-        if (is_first_nan || curr_val < min_val) {
+        if (sycl::isnan(curr_val)) return logical_i;
+        if (curr_val < min_val) {
             min_idx = logical_i;
             min_val = curr_val;
         }
@@ -102,12 +105,13 @@ int cblas_izamin(int n, const std::complex<double> *x, int incx) {
     }
     int min_idx = 0;
     auto min_val = abs_val(x[0]);
+    if (sycl::isnan(min_val)) return 0;
 
-    for (int logical_i = 0; logical_i < n; ++logical_i) {
+    for (int logical_i = 1; logical_i < n; ++logical_i) {
         int i = logical_i * std::abs(incx);
         auto curr_val = abs_val(x[i]);
-        bool is_first_nan = std::isnan(curr_val) && !std::isnan(min_val);
-        if (is_first_nan || curr_val < min_val) {
+        if (sycl::isnan(curr_val)) return logical_i;
+        if (curr_val < min_val) {
             min_idx = logical_i;
             min_val = curr_val;
         }

--- a/tests/unit_tests/blas/include/reference_blas_templates.hpp
+++ b/tests/unit_tests/blas/include/reference_blas_templates.hpp
@@ -1461,12 +1461,14 @@ int iamin(const int *n, const float *x, const int *incx) {
     }
     int min_idx = 0;
     auto min_val = abs_val(x[0]);
-    if (sycl::isnan(min_val)) return 0;
+    if (sycl::isnan(min_val))
+        return 0;
 
     for (int logical_i = 1; logical_i < *n; ++logical_i) {
         int i = logical_i * std::abs(*incx);
         auto curr_val = abs_val(x[i]);
-        if (sycl::isnan(curr_val)) return logical_i;
+        if (sycl::isnan(curr_val))
+            return logical_i;
         if (curr_val < min_val) {
             min_idx = logical_i;
             min_val = curr_val;
@@ -1482,12 +1484,14 @@ int iamin(const int *n, const double *x, const int *incx) {
     }
     int min_idx = 0;
     auto min_val = abs_val(x[0]);
-    if (sycl::isnan(min_val)) return 0;
+    if (sycl::isnan(min_val))
+        return 0;
 
     for (int logical_i = 1; logical_i < *n; ++logical_i) {
         int i = logical_i * std::abs(*incx);
         auto curr_val = abs_val(x[i]);
-        if (sycl::isnan(curr_val)) return logical_i;
+        if (sycl::isnan(curr_val))
+            return logical_i;
         if (curr_val < min_val) {
             min_idx = logical_i;
             min_val = curr_val;
@@ -1503,12 +1507,14 @@ int iamin(const int *n, const std::complex<float> *x, const int *incx) {
     }
     int min_idx = 0;
     auto min_val = abs_val(x[0]);
-    if (sycl::isnan(min_val)) return 0;
+    if (sycl::isnan(min_val))
+        return 0;
 
     for (int logical_i = 1; logical_i < *n; ++logical_i) {
         int i = logical_i * std::abs(*incx);
         auto curr_val = abs_val(x[i]);
-        if (sycl::isnan(curr_val)) return logical_i;
+        if (sycl::isnan(curr_val))
+            return logical_i;
         if (curr_val < min_val) {
             min_idx = logical_i;
             min_val = curr_val;
@@ -1524,12 +1530,14 @@ int iamin(const int *n, const std::complex<double> *x, const int *incx) {
     }
     int min_idx = 0;
     auto min_val = abs_val(x[0]);
-    if (sycl::isnan(min_val)) return 0;
+    if (sycl::isnan(min_val))
+        return 0;
 
     for (int logical_i = 1; logical_i < *n; ++logical_i) {
         int i = logical_i * std::abs(*incx);
         auto curr_val = abs_val(x[i]);
-        if (sycl::isnan(curr_val)) return logical_i;
+        if (sycl::isnan(curr_val))
+            return logical_i;
         if (curr_val < min_val) {
             min_idx = logical_i;
             min_val = curr_val;

--- a/tests/unit_tests/blas/include/reference_blas_templates.hpp
+++ b/tests/unit_tests/blas/include/reference_blas_templates.hpp
@@ -1461,12 +1461,13 @@ int iamin(const int *n, const float *x, const int *incx) {
     }
     int min_idx = 0;
     auto min_val = abs_val(x[0]);
+    if (sycl::isnan(min_val)) return 0;
 
-    for (int logical_i = 0; logical_i < *n; ++logical_i) {
+    for (int logical_i = 1; logical_i < *n; ++logical_i) {
         int i = logical_i * std::abs(*incx);
         auto curr_val = abs_val(x[i]);
-        bool is_first_nan = std::isnan(curr_val) && !std::isnan(min_val);
-        if (is_first_nan || curr_val < min_val) {
+        if (sycl::isnan(curr_val)) return logical_i;
+        if (curr_val < min_val) {
             min_idx = logical_i;
             min_val = curr_val;
         }
@@ -1481,12 +1482,13 @@ int iamin(const int *n, const double *x, const int *incx) {
     }
     int min_idx = 0;
     auto min_val = abs_val(x[0]);
+    if (sycl::isnan(min_val)) return 0;
 
-    for (int logical_i = 0; logical_i < *n; ++logical_i) {
+    for (int logical_i = 1; logical_i < *n; ++logical_i) {
         int i = logical_i * std::abs(*incx);
         auto curr_val = abs_val(x[i]);
-        bool is_first_nan = std::isnan(curr_val) && !std::isnan(min_val);
-        if (is_first_nan || curr_val < min_val) {
+        if (sycl::isnan(curr_val)) return logical_i;
+        if (curr_val < min_val) {
             min_idx = logical_i;
             min_val = curr_val;
         }
@@ -1501,12 +1503,13 @@ int iamin(const int *n, const std::complex<float> *x, const int *incx) {
     }
     int min_idx = 0;
     auto min_val = abs_val(x[0]);
+    if (sycl::isnan(min_val)) return 0;
 
-    for (int logical_i = 0; logical_i < *n; ++logical_i) {
+    for (int logical_i = 1; logical_i < *n; ++logical_i) {
         int i = logical_i * std::abs(*incx);
         auto curr_val = abs_val(x[i]);
-        bool is_first_nan = std::isnan(curr_val) && !std::isnan(min_val);
-        if (is_first_nan || curr_val < min_val) {
+        if (sycl::isnan(curr_val)) return logical_i;
+        if (curr_val < min_val) {
             min_idx = logical_i;
             min_val = curr_val;
         }
@@ -1521,12 +1524,13 @@ int iamin(const int *n, const std::complex<double> *x, const int *incx) {
     }
     int min_idx = 0;
     auto min_val = abs_val(x[0]);
+    if (sycl::isnan(min_val)) return 0;
 
-    for (int logical_i = 0; logical_i < *n; ++logical_i) {
+    for (int logical_i = 1; logical_i < *n; ++logical_i) {
         int i = logical_i * std::abs(*incx);
         auto curr_val = abs_val(x[i]);
-        bool is_first_nan = std::isnan(curr_val) && !std::isnan(min_val);
-        if (is_first_nan || curr_val < min_val) {
+        if (sycl::isnan(curr_val)) return logical_i;
+        if (curr_val < min_val) {
             min_idx = logical_i;
             min_val = curr_val;
         }

